Homework3/Stack/Stack.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Stack
+{
+    public class Stack<T> : IStack<T>, IEnumerable
+    {
+        public T[] Items { get; set; } = new T[0];
+        public int Size { get; set; }   
+
+
+        public void Push(T value)
+        {
+            if (Size == Items.Length)
+            {
+                int newlength = Size == 0 ? 1 : Size + 1;
```

Comment:
Dynamic resizing will be called each time when new item is added, it is not optimal. 

Instead, initialize items with prefedined size, for example 16. 
When item is out of range, throw exception.